### DISCUSSION TITLE
nginx and opentelemetry-webserver-sdk  :  unable to start nginx due to error  "libopentelemetry_webserver_sdk.so: cannot open shared object file: No such file or directory"

### DISCUSSION THREAD

**Discussion by PsychicMushrooms** [open-telemetry/community#1646] [Community Discussions]
Hi  all,  
I'm trying to get nginx 1.22.1 started on centos 8 in order to test the  Nginx Webserver Module as described here which connects to the Otel Collector:
https://github.com/open-telemetry/opentelemetry-cpp-contrib/tree/main/instrumentation/otel-webserver-module#platform-supported-1
but keep getting this error :
**nginx[1470]: nginx: [emerg] dlopen() "/opt/opentelemetry-webserver-sdk/WebServerModule/Nginx/1.23.1/ngx_http_opentelemetry_module.so" failed (libopentelemetry_webserver_sdk.so: cannot open shared object file: No such file or directory) in /etc/nginx/nginx.conf:9**

I have exported the  LD_LIBRARY_PATH environment variable and /opt/opentelemetry-webserver-sdk has permissions 777 set recursively too,  but still no joy.
ie.
export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/opt/opentelemetry-webserver-sdk/sdk_lib/lib
[nginx.conf.txt](https://github.com/open-telemetry/community/files/12341596/nginx.conf.txt)

**Comment by trask** [comment]
hi @PsychicMushrooms! I'd recommend asking in https://github.com/open-telemetry/opentelemetry-cpp-contrib or in [#otel-cpp](https://cloud-native.slack.com/archives/C01N3AT62SJ) slack channel

**Comment by PsychicMushrooms** [comment]
thanks.
